src/sc/sidechain.h

```diff
@@ -170,10 +172,13 @@ class CSidechain {
         READWRITE(balance);
         READWRITE(fixedParams);
         READWRITE(mImmatureAmounts);
-        READWRITE(scFees);
+        if (isNonCeasing())
+            READWRITE(scFees_v2);
+        else
+            READWRITE(scFees);
         if (ser_action.ForRead())
         {
-            if (!scFees.empty())
+            if (!scFees.empty() || !scFees_v2.empty())
             {
                 maxSizeOfScFeesContainers = getMaxSizeOfScFeesContainers();
             }
```

Comment:
What is this for ?
I thought a serialization op was not meant to change the state of the object.
Why we do change `maxSizeOfScFeesContainers` and, btw, we dont' serialize it ?

Reply:
This looks super weird and caused an issue during tests, as I originally forgot to check non-emptiness of `scFees_v2`.

It seems that `maxSizeOfScFeesContainers` is not serialized, hence it must be recovered and re-evaluated when a node is reactivated after a shutdown, as this variable is initialized with a proper value only when a sc creation transaction is received (see `CSidechain::InitScFees()`).

Given this context, it is possible that `scFees` are correctly emplaced in the lists when deserializing a new block, but `maxSizeOfScFeesContainers` is still set as -1.

Reply:
So let me understand ... we're using serialization here as a constructor ?